### ISSUE TITLE
Icon parsing with template matching

### ISSUE THREAD

**Issue by SammysHP** [cgeo/cgeo#6204]
As described in #6175 we should try template matching to detect the cache type.

1. Find bounding box (easy because of transparent background)
1. Calculate center
1. Template matching, select best match

This is more fuzzy than you current solution (comparing single pixel color), uses shape/texture (not only color) that should detect all cache types and it should also work with partially overlapped icons. Optimizations (generalized feature vector) are possible. Anyway the parsing should be much faster than downloading the tiles.

**Comment by Lineflyer** [comment]
Could this also help in determining the cache coordinates?

**Comment by SammysHP** [comment]
That depends on the accuracy of the icons and the detection quality of the center (overlapping icons). Another possibility would be to use the coarse coordinates and shift the template to find the best match.

**Comment by mucek4** [comment]
Currently we look only for color. We have RGB from 0-255 and parse that really quick. The problem is similar colors.
Template matching works good until caches overlap. As soon a template is overlapped we have a problem.
I have tried my theory: use the same type of parsing, but instead of colors use edges. I made grayscale image and compared shapes. results are really good, however I have some problems with white edges around the image. They are all the same at multiple caches. Will see how it will go.
